src/wasm-lib/kcl/src/std/shapes.rs

```diff
@@ -66,28 +78,71 @@ async fn inner_circle(
         SketchSurfaceOrGroup::SketchSurface(surface) => surface,
         SketchSurfaceOrGroup::SketchGroup(group) => group.on,
     };
-    let mut sketch_group = crate::std::sketch::inner_start_profile_at(
-        [center[0] + radius, center[1]],
+    let sketch_group = crate::std::sketch::inner_start_profile_at(
+        [data.center[0] + data.radius, data.center[1]],
         sketch_surface,
         None,
         exec_state,
         args.clone(),
     )
     .await?;
 
-    // Call arc.
-    sketch_group = crate::std::sketch::inner_arc(
-        crate::std::sketch::ArcData::AnglesAndRadius {
-            angle_start: 0.0,
-            angle_end: 360.0,
-            radius,
+    let angle_start = Angle::from_degrees(0.0);
+    let angle_end = Angle::from_degrees(360.0);
+
+    if angle_start == angle_end {
+        return Err(KclError::Type(KclErrorDetails {
+            message: "Arc start and end angles must be different".to_string(),
+            source_ranges: vec![args.source_range],
+        }));
+    }
+
+    let id = uuid::Uuid::new_v4();
+
+    args.batch_modeling_cmd(
+        id,
+        ModelingCmd::ExtendPath {
+            path: sketch_group.id,
+            segment: kittycad::types::PathSegment::Arc {
+                start: angle_start,
+                end: angle_end,
+                center: data.center.into(),
+                radius: data.radius,
+                relative: false,
+            },
+        },
+    )
+    .await?;
+
+    let current_path = Path::Circle {
+        base: BasePath {
+            from: data.center,
+            to: data.center,
+            tag: tag.clone(),
+            geo_meta: GeoMeta {
+                id,
+                metadata: args.source_range.into(),
+            },
+        },
+        radius: data.radius,
+        center: data.center,
+        ccw: angle_start.degrees() < angle_end.degrees(),
+    };
+
+    let mut new_sketch_group = sketch_group.clone();
+    if let Some(tag) = &tag {
+        new_sketch_group.add_tag(tag, &current_path);
+    }
+
+    new_sketch_group.value.push(current_path);
+
+    args.batch_modeling_cmd(
+        id,
+        ModelingCmd::ClosePath {
+            path_id: new_sketch_group.id,
         },
-        sketch_group,
-        tag,
-        args.clone(),
     )
     .await?;
 
-    // Call close.
-    crate::std::sketch::inner_close(sketch_group, None, args).await
```

Comment:
Using inner_close was a no go as well, because while we do want send the close command to the engine, calling this adds a new segment to the sketch group which doesn't make sense, circle is just one segment.